### DISCUSSION TITLE
pybricks 3.5.0 on ev3

### DISCUSSION THREAD

**Discussion by Nikolay-Zagrebin** [pybricks/support#1653] [Q&A]
Share your experience, who managed to run pybricks 3.5.0 on ev3?

Earlier, David wrote that it is possible to install pybricks-micropython3 from the Debian repository, but there is only version 3.3.0b6.

In principle, pybricks 3.5.0 can be installed from the pip repository "pip install pybricks" but I can't use this package in the python project, or I just don't know how to do it))

**Comment by Nikolay-Zagrebin** [comment]
Doesn't anyone seem to know?
I'll have to clean it up myself))

**Comment by dlech** [comment]
There is a long discussion at https://github.com/orgs/pybricks/discussions/1171

The main conclusion is that motor control is broken because of changes made in v3 that made things better on Powered Up hubs but don't work so well with Linux on EV3.

**Comment by BertLindeman** [comment]
Hi Nokolay,

If after what David posted above, and read through that discussion, and you still want to try a 3.5 version, you can without damaging the current version you use.

My way:
From CI-build 3357 (CI for Continuous Integration I think) I downloaded the pybricks-micropyhon zip file and copied the program in that zip to a directory in my path on the EV3 (Still running as a Linux server).

- Downloaded from [pybricks-micropython](https://github.com/pybricks/pybricks-micropython/actions/runs/8556042869) 
- unzipped to my path and renamed to ```pybricks-micropython_build_3357``` so nothing gets overwritten
- checked with:
```python

#!/usr/bin/env pybricks-micropython_build_3357

from pybricks import version

print("1:", version)
import usys
print("2:", usys.implementation)
print("3:", usys.version)
print("4:", usys.version_info)
```

Results in:
```
1: ('ev3', '3.5.0b1', 'v1.20.0-23-g6c633a8dd on 2024-04-04')
2: (name='micropython', version=(1, 20, 0), _machine='LEGO MINDSTORMS EV3 Intelligent Brick with Texas Instruments AM1808', _mpy=3334)
3: 3.4.0; Pybricks MicroPython v1.20.0-23-g6c633a8dd on 2024-04-04
4: (3, 4, 0)
```

Be sure you do not overwrite anything in this.

Bert

**Comment by Nikolay-Zagrebin** [comment]
Thank you guys very much!!!
You are the best!!!
With best wishes from Russia!!
